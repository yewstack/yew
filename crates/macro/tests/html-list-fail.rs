use yew_macro::{html, test_html};

test_html! { |t1| <> }
test_html! { |t2| </> }
test_html! { |t3| <><> }
test_html! { |t4| </></> }
test_html! { |t5| <><></> }
test_html! { |t6| <></><></> }
test_html! { |t7| <>invalid</> }

fn main() {}
