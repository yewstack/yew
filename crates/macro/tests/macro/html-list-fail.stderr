error: this open tag has no corresponding close tag
 --> $DIR/html-list-fail.rs:4:13
  |
4 |     html! { <> };
  |             ^^

error: this close tag has no corresponding open tag
 --> $DIR/html-list-fail.rs:5:13
  |
5 |     html! { </> };
  |             ^^^

error: this open tag has no corresponding close tag
 --> $DIR/html-list-fail.rs:6:13
  |
6 |     html! { <><> };
  |             ^^

error: this close tag has no corresponding open tag
 --> $DIR/html-list-fail.rs:7:13
  |
7 |     html! { </></> };
  |             ^^^

error: this open tag has no corresponding close tag
 --> $DIR/html-list-fail.rs:8:13
  |
8 |     html! { <><></> };
  |             ^^

error: only one root html element allowed
 --> $DIR/html-list-fail.rs:9:18
  |
9 |     html! { <></><></> };
  |                  ^^^^^

error: expected valid html element
  --> $DIR/html-list-fail.rs:10:15
   |
10 |     html! { <>invalid</> };
   |               ^^^^^^^

error: unexpected end of input, expected expression
  --> $DIR/html-list-fail.rs:11:5
   |
11 |     html! { <key=></>}
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: this open tag has no corresponding close tag
  --> $DIR/html-list-fail.rs:12:13
   |
12 |     html! { <key="key".to_string()>invalid</key>}
   |             ^^^^^^^^^^^^^^^^^^^^^^^
